### PR TITLE
mad: run autoreconf before configure.

### PR DESCRIPTION
**Pull request by mahiuchun**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This would fix the arm build error mentioned in https://github.com/Homebrew/brew/issues/7857

The lack of test is pre-existing.
EDIT: A test is added now.

### CODE REVIEW

**Comment by SMillerDev** [comment]
You should add a test so we can be sure this doesn't break anything.

**Comment by mahiuchun** [comment]
Done. Do I need to do anything about "missing license"?

File: Formula/mad.rb [hunk 1]
```diff
@@ -22,12 +22,254 @@ class Mad < Formula
...
+    touch "NEWS", "AUTHORS", "ChangeLog"
+    system "autoreconf", "-fiv"
     system "./configure", "--disable-debugging", "--enable-fpm=64bit", "--prefix=#{prefix}"
     system "make", "CFLAGS=#{ENV.cflags}", "LDFLAGS=#{ENV.ldflags}", "install"
     (lib+"pkgconfig/mad.pc").write pc_file
   end
 
+  test do
+    testpath.install resource("sfx.mp3")
+    (testpath/"minimad.c").write <<~EOS
```
Comment (SMillerDev) [inline Formula/mad.rb:44]: Can you download the file as a resource too?

**Comment by SMillerDev** [comment]
If you know the license you can add it, otherwise it's fine.

File: Formula/mad.rb (hunk 1 above)
Comment (mahiuchun) [reply Formula/mad.rb:44]: It is from `libmad` tarball but I cannot find an official place to download it as single file. Is there a special way to refer to the source or should I just re-download the tarball and extract from there?

Comment (fxcoudert) [reply Formula/mad.rb:45]: We can copy it in the `install` method, like this:
`pkgshare.install "examples/wherever/minimad.c"`

Comment (mahiuchun) [reply Formula/mad.rb]: Done. I also fixed the touch error and switched to using test.mp3 in test_fixtures to reduce complexity of the test.

File: Formula/mad.rb [hunk 2]
```diff
@@ -22,10 +23,25 @@ class Mad < Formula
...
   end
 
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+
   def install
+    touch "NEWS"
+    touch "AUTHORS"
+    touch "ChangeLog"
```
Comment (fxcoudert) [inline Formula/mad.rb:33]: Why is that necessary?

Comment (mahiuchun) [reply Formula/mad.rb:33]: `autoreconf -fiv` would not succeed otherwise

**Review by SMillerDev — APPROVED** [review]
Thanks @mahiuchun ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉
